Source/AwsCommonRuntimeKit/auth/signing/SigningConfig.swift

```diff
@@ -2,121 +2,233 @@
 //  SPDX-License-Identifier: Apache-2.0.
 
 import AwsCAuth
+import Foundation
 
-// TODO: verify callback logic, fix pointers, and maybe error handling
-public struct SigningConfig: CStruct {
-    public typealias ShouldSignHeader = (String) -> Bool
+public struct SigningConfig: CStructWithUserData {
```

Comment:
related to my earlier comment about naming the signer...

if we name it AwsSigner, everything in this file should probably also get an "AWS" prefix
or we leave the prefix off and the current names are fine

Reply:
Thanks, I have not changed it for now.